File: Konfigurator-nestjs-Backend/konfigurator-nestjs-backend/src/user-management/user-management.controller.ts
import { Controller, Post, Body, Res } from '@nestjs/common';
import { UserManagementService } from './user-management.service';
import { GetUserRequestDto } from 'src/dto/get-UserRequest.dto';
import { Response } from 'express';


@Controller('UserManagement')
export class UserManagementController {
  constructor(private readonly userManagementService: UserManagementService) {}

  @Post('add')
  async CreateUser(@Res() response: Response, @Body() request: GetUserRequestDto) {
    const resp = await this.userManagementService.Createuser(request);

    if(resp.success) {
      return response.status(200).send('"' + resp.message + '"');
    }
    else 
    {
      return response.status(400).send(resp.message)
    }
  }

  @Post('login')
  async LoginUser(@Res() response: Response, @Body() request: GetUserRequestDto) {
    const resp = await this.userManagementService.LoginUser(request);
    
    if(resp.success) {
      return response.send(resp.data);
    }
    
    return response.status(400).send(resp.message)
  }
}

File: Konfigurator-aspnetcore-Backend/Konfigurator-aspnetcore-Backend/Controllers/UserManagementController.cs
using Konfigurator_aspnetcore_Backend.Data;
using Konfigurator_aspnetcore_Backend.Models.DTOs;
using Konfigurator_aspnetcore_Backend.Services.Configuration;
using Konfigurator_aspnetcore_Backend.Services.UserManagement;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Konfigurator_aspnetcore_Backend.Controllers
{
    [Route("[controller]")]
    [ApiController]
    public class UserManagementController : ControllerBase
    {
        private readonly IUserManagementService _userManagementService;

        public UserManagementController(IUserManagementService userManagementService)
        {
            _userManagementService = userManagementService;
        }

        [HttpPost("add")]
        public async Task<ActionResult<ServiceResponseDto<string>>> CreateUser(UserRequestDto request)
        {
            var response = await _userManagementService.CreateUser(request);
            if (response.Success)
                return Ok(response.Message);
            return BadRequest(response.Message);
        }

        [HttpPost("login")]
        public async Task<ActionResult<ServiceResponseDto<UserResponseDto>>> LoginUser(UserRequestDto request)
        {
            var response = await _userManagementService.LoginUser(request);
            if (response.Success)
                return Ok(response.Data);
            return BadRequest(response.Message);
        }
    }
}
